mesonpy/__init__.py

```diff
@@ -133,40 +133,32 @@ def _init_colors() -> Dict[str, str]:
 assert all(re.match(_EXTENSION_SUFFIX_REGEX, x) for x in _EXTENSION_SUFFIXES)
 
 
-# Maps wheel installation paths to Meson installation path placeholders.
+# Map Meson installation path placeholders to wheel installation paths.
 # See https://docs.python.org/3/library/sysconfig.html#installation-paths
-_SCHEME_MAP = {
-    'scripts': ('{bindir}',),
-    'purelib': ('{py_purelib}',),
-    'platlib': ('{py_platlib}', '{moduledir_shared}'),
-    'headers': ('{includedir}',),
-    'data': ('{datadir}',),
-    # our custom location
-    'mesonpy-libs': ('{libdir}', '{libdir_shared}')
+_INSTALLATION_PATH_MAP = {
+    '{bindir}': 'scripts',
+    '{py_purelib}': 'purelib',
+    '{py_platlib}': 'platlib',
+    '{moduledir_shared}': 'platlib',
+    '{includedir}': 'headers',
+    '{datadir}': 'data',
+    # custom location
+    '{libdir}': 'mesonpy-libs',
+    '{libdir_shared}': 'mesonpy-libs',
 }
 
 
-def _map_meson_destination(destination: str) -> Tuple[str, pathlib.Path]:
-    """Map a Meson installation path to a wheel installation location.
-
-    Return a (wheel path identifier, subpath inside the wheel path) tuple.
-
-    """
-    parts = pathlib.Path(destination).parts
-    for folder, placeholders in _SCHEME_MAP.items():
-        if parts[0] in placeholders:
-            return folder, pathlib.Path(*parts[1:])
-    raise BuildError(f'Could not map installation path to an equivalent wheel directory: {destination!r}')
-
-
 def _map_to_wheel(sources: Dict[str, Dict[str, Any]]) -> DefaultDict[str, List[Tuple[pathlib.Path, str]]]:
     """Map files to the wheel, organized by wheel installation directrory."""
     wheel_files = collections.defaultdict(list)
     for group in sources.values():
         for src, target in group.items():
-            directory, path = _map_meson_destination(target['destination'])
-            if directory is not None:
-                wheel_files[directory].append((path, src))
+            dest = target['destination']
+            parts = pathlib.Path(dest).parts
+            path = _INSTALLATION_PATH_MAP.get(parts[0])
+            if path is None:
+                raise BuildError(f'Could not map installation path to an equivalent wheel directory: {dest!r}')
+            wheel_files[path].append((pathlib.Path(*parts[1:]), src))
```

Comment:
I think choosing the variables a bit more carefully makes the code easier to read. Also, using `destination.relative_to(anchor)` instead of `pathlib.Path(*parts[1:])`, for eg., makes what we are trying to do a bit more explicit and quicker to understand.

```suggestion
            destination = pathlib.Path(target['destination'])
            anchor = destination.parts[0]
            scheme = _INSTALLATION_PATH_MAP.get(anchor)
            if scheme is None:
                raise BuildError(f'Could not map installation path to an equivalent wheel directory: {destination!r}')
            wheel_files[scheme].append((destination.relative_to(anchor), src))
```

Reply:
`destination.relative_to(anchor)` is wrong and does not work.

We are dealing with `sysconfig` concepts. In this context a "scheme" is the collections of concrete directories where the parts of a wheel are installed. I've even recently added a link to the documentation explaining this. As the `sysconfig` maintainer I'm surprised that you get the basic nomenclature wrong.